File: problems/problems_1846/solution.py
import solution


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.maximumElementAfterDecrementingAndRearranging(list(test_input))

    def maximumElementAfterDecrementingAndRearranging(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        n = len(arr)
        cnts = [0] * (n + 1)
        for num in arr:
            cnts[min(num, n)] += 1
        ans = 0
        for i in range(1, n + 1):
            ans = min(i, ans + cnts[i])
        return ans

File: problems/problems_1846/Solution.java
package problems.problems_1846;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int maximumElementAfterDecrementingAndRearranging(int[] arr) {
        
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] arr = jsonArrayToIntArray(inputJsonValues[0]);
        return JSON.toJSON(maximumElementAfterDecrementingAndRearranging(arr));
    }
}
